puma/plot_base.py

```diff
@@ -397,21 +400,38 @@ def set_title(self, title: str = None, **kwargs):
         """
         self.axis_top.set_title(self.title if title is None else title, **kwargs)
 
-    def set_logy(self, force: bool = False):
-        """Set log scale of y-axis of main panel.
+    def set_log(self, force: bool = False):
+        """Set log scale of the axes. For the y-axis, only the main panel is
+        set. For the x-axes (also from the ratio subpanels), all are changed.
 
         Parameters
         ----------
         force : bool, optional
-            Forcing logy even if class variable is False, by default False
+            Forcing log even if class variable is False, by default False
         """
-        if not self.logy and not force:
-            return
-        if not self.logy:
-            logger.warning("Setting log of y-axis but `logy` flag was set to False.")
-        self.axis_top.set_yscale("log")
-        ymin, ymax = self.axis_top.get_ylim()
-        self.y_scale = ymin * ((ymax / ymin) ** self.y_scale) / ymax
+
+        if self.logx or force:
+            if not self.logx and force:
```

Comment:
```suggestion
            if not self.logx:
```
this should be enough